### ISSUE TITLE
Error: No such file or directory - /tmp/logstash

### ISSUE THREAD

**Issue by webteb** [logstash-plugins/logstash-input-s3#32]
doesn't seem to work at all, i keep getting this error. (logstash version 1.5.0-rc3)
with version logstash 1.4.2 it works but i get errors about utf-8.
running with this configuration:
input {
        s3 {
                bucket => "bucketname"
                region => "region"
		access_key_id => "key"
		secret_access_key => "secret"
		type => "elb"
                prefix => "elb/www/AWSLogs/1234567890/elasticloadbalancing/region/2015/04/"
                interval => 60
        }
}
any ideas ? 
thanks.

**Comment by markwalkom** [comment]
What's the full error you are seeing.

**Comment by webteb** [comment]
A plugin had an unrecoverable error. Will restart this plugin.
  Plugin: <LogStash::Inputs::S3 bucket=>"awslogs.webteb", region=>"eu-west-1", access_key_id=>"keyid**", secret_access_key=>"secret**", type=>"elb", prefix=>"elb/www/AWSLogs/1234567890/elasticloadbalancing/eu-west-1/2015/05/", temporary_directory=>"/tmp/logstash">
  Error: No such file or directory - /tmp/logstash/1234567890_elasticloadbalancing_eu-west-1_www-elb_20150501T0000Z_54.76.69.58_zqsz0fa1.log {:level=>:error}

**Comment by webteb** [comment]
i started logstash with --debug flag and got additional information that the error is happening at line 300 where this is printed: "S3 input: Download remove file",  :line=>"300", :method=>"download_remote_file"}

**Comment by webteb** [comment]
@markwalkom can you help me with this ?
thanks

**Comment by webteb** [comment]
i found the solution,
i manually created the directory /tmp/logstash and set permissions for current user with chown

**Comment by ph** [comment]
Reopening, this should work out of the box with the default temp directory.

**Comment by ph** [comment]
@webteb Also you can use the `temporary_directory` option in your config to specify a directory.

**Comment by webteb** [comment]
ok, i will try that, you should know that i am running logstash like this:
bin/logstash agent -f my.conf -v
and am not using root, normal user.

thanks.

**Comment by vanga** [comment]
I also ran into this problem, creating /tmp/logstash manually was enought.

@ph I have made some observation, that the plugin doesn't seem to be deleting the files from the temporary location after it's done with the processing. After some there will be too many files and then the logstash process gets killed by OOM killer. I am not sure how having too many files in temp location has got anything to do with RAM.
